### DISCUSSION TITLE
dateTime in SADL

### DISCUSSION THREAD

**Discussion by AbhaMoitra** [SemanticApplicationDesignLanguage/sadl#613] [General]
@crapo : I am using dateTime in a SADL model and then I query the dateTime to generate a csv file. I am not sure I understand what is going on; here are 2 examples I tried out following examples in https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html#ISO_OFFSET_DATE_TIME

Example 1: in SADL
     has generatedAtTime "2017-03-23T10:03:16"
In OWL file
    <j.0:generatedAtTime rdf:datatype="http://www.w3.org/2001/XMLSchema#dateTime"
    >2017-03-22T20:00:00-04:00</j.0:generatedAtTime>
In csv file 
    Wed Mar 22 20:00:00 EDT 2017
 
Example2: in SADL 
    has generatedAtTime "2017-03-23T10:03:16+04:00"
In OWL file
    <j.0:generatedAtTime rdf:datatype="http://www.w3.org/2001/XMLSchema#dateTime"
    >2017-03-22T23:16:00-04:00</j.0:generatedAtTime>
In csv file
    Wed Mar 22 23:16:00 EDT 2017

**Comment by AbhaMoitra** [comment]
It seems I should replace "T" by space in 
   has generatedAtTime "2017-03-23T10:03:16"

**Reply by crapo** [reply]
That is correct. You can also include a timezone or UTC. SADL uses pojava DateTime, see https://javadoc.io/static/org.pojava/org.pojava.datetime/3.0.0/index.html?org/pojava/datetime/DateTime.html.

**Reply by crapo** [reply]
Note that the org.pojava version that we are using isn't the latest available. If you encounter any issues please let me know.

**Reply by crapo** [reply]
@AbhaMoitra , I've added dateTime to those datatype property value suggestions for content assist. When this is pushed and the PR created and merged, you would get this from content assist: 
         
      with generatedAtTime "MM/DD/YYYY hh:mm:ss PM TZD"

which might, for example, be filled in with (other formats accepted, this serves only as an example)
      
      with generatedAtTime "2/3/2021 7:35 PM EST"

which would translate to

     <datetimeconversion:generatedAtTime rdf:datatype="http://www.w3.org/2001/XMLSchema#dateTime"
     >2021-02-03T19:35:00-05:00</datetimeconversion:generatedAtTime>

.

**Comment by AbhaMoitra** [comment]
@crapo - thank you. For our project we are planning to use UTC and we have it working.
